src/components/OSCALSystemImplementationInventoryItems.js

```diff
@@ -0,0 +1,167 @@
+import TableContainer from "@mui/material/TableContainer";
+import TableHead from "@mui/material/TableHead";
+import TableRow from "@mui/material/TableRow";
+
+const SmallTableCell = styled(TableCell)`
+  text-align: right;
+  padding: 0.75em 0.75em;
+`;
+
+const OSCALSystemImplementationTableTitle = styled(Typography)`
+  flex: 1 1 100%;
+  padding-top: 1em;
+  padding-bottom: 1em;
+`;
+
+const StyledHeaderTableCell = styled(TableCell)(({ theme }) => ({
+  [`&.${tableCellClasses.head}`]: {
+    backgroundColor: theme.palette.primary.main,
+    color: theme.palette.common.white,
+  },
+  textAlign: "left",
+  minWidth: "10em",
+}));
+
+const StyledTableRow = styled(TableRow)(({ theme }) => ({
+  // Use hover color for even numbered rows
+  "&:nth-of-type(even)": {
+    backgroundColor: theme.palette.action.hover,
+  },
+}));
+
+function PropertiesTable(props) {
+  return (
+    <TableContainer>
+      <Table size="small">
+        <TableBody>
+          {props.list?.map((property) => (
+            <TableRow key={property.name}>
+              <TableCell component="th" scope="row">
+                {property.name}
+              </TableCell>
+              <SmallTableCell>{property.value}</SmallTableCell>
+            </TableRow>
+          ))}
+        </TableBody>
+      </Table>
+    </TableContainer>
+  );
+}
+
+function OSCALResponsibleParties(props) {
+  const getPartyName = (partyUuid) =>
+    props.parties?.find((party) => party.uuid === partyUuid)?.name;
+
+  return (
+    <TableContainer>
+      <Table size="small">
+        <TableBody>
+          {props.responsibleParties?.map((party) => (
+            <TableRow key={party["role-id"]}>
+              <TableCell component="th" scope="row">
+                {party["role-id"]}
+              </TableCell>
+              <TableCell align="right">
+                {party["party-uuids"] &&
+                  party["party-uuids"].map((partyUuid) =>
+                    getPartyName(partyUuid)
+                  )}
+              </TableCell>
+            </TableRow>
+          ))}
+        </TableBody>
+      </Table>
+    </TableContainer>
+  );
+}
+
+function ImplementedComponents(props) {
+  const getComponent = (compUuid) =>
+    props.components?.find((component) => component.uuid === compUuid);
+
+  return (
+    <TableContainer>
+      <Table size="small">
+        <TableBody>
+          {props.implementedComponents?.map((comp) => (
+            <TableRow key={comp["component-uuid"]}>
+              <TableCell
+                component="th"
+                scope="row"
+                aria-label={`${comp["component-uuid"]} implemented component`}
+              >
+                {getComponent(comp["component-uuid"])?.title}
+              </TableCell>
+              <TableCell align="right">
+                {getComponent(comp["component-uuid"])?.type}
+              </TableCell>
+            </TableRow>
+          ))}
```

Comment:
Let's minimize the number of lookups. Comments inline are for explanation of review, not necessarily meant to be kept in the submitted code.

```suggestion
          {props.implementedComponents
            // This custom object lets us store the UUID while also performing
            // the lookup for each component only once
            ?.map((comp) => ({
              uuid: comp["component-uuid"],
              component: getComponent(comp["component-uuid"])
            }))
            // `comp` is probably not the best thing here but i dunno
            ?.map((comp) => (
            <TableRow key={comp.uuid}>
              <TableCell
                component="th"
                scope="row"
                aria-label={`${comp.uuid} implemented component`}
              >
                {comp.component?.title}
              </TableCell>
              <TableCell align="right">
                {comp.component?.type}
              </TableCell>
            </TableRow>
          ))}
```

Reply:
Great suggestion! I'll go ahead and commit this change. Perhaps we keep the second mapped `comp` for now and iterate on later? I think it would be a good idea to note this.

Reply:
Just tested it and it works well!